Watchdog handling cleanup

pennam opened the pull request:
Aim of this pr is to cleanup watchdog handling replacing architecture specific function calls with an unique implementation that actually takes care of the architecture.

I've added also an out of scope commit that fixes the declared NANO_RP2040 maximum watchdog timeout.

I didn't change the `mbed_watchdog_trigger_reset` function with the hope to find a better fix in the future and remove it.

I'm opening as a draft to start testing.

| Board  | Result |
| ------------- | ------------- |
| NANO RP2040  | :heavy_check_mark:|
| PORTENTA H7  |   |
| NANO 33 IoT |:heavy_check_mark: |
| MKR WIFI 1010 |:heavy_check_mark: |

aentinger reviewed (approved):
Great work! I :heart: it.

aentinger commented:
If the tests work, its good to merge :+1:

pennam commented:
Thanks, i will try to add the test results as soon as possible. I'm still having troubles connecting to create.arduino.cc

pennam commented:
I've added a commit  that should take care of your changes here https://github.com/arduino-libraries/WiFiNINA/pull/182/files
I was unsure on how to handle library compatibility so i've added this define WIFI_HAS_FEED_WATCHDOG_FUNC to make sure to not break anything. Doing this way it needs also to be added into the WiFi NINA library https://github.com/pennam/WiFiNINA/commit/67800139db83c621bd5bf51cad98242bcdbd86d8

I did not remove this additional watchdog feed https://github.com/arduino-libraries/ArduinoIoTCloud/blob/d09c6dc3c6163f310118f9342a0f50df2424aaf8/src/ArduinoIoTCloudTCP.cpp#L323-L331 otherwise i still get some resets.
